### PR TITLE
Export InvalidIndexError

### PR DESCRIPTION
**Pull request by horta**
- [x] closes #xxxx (there is no such an issue)
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
@horta  there is a `pandas.errors` submodule where this could be put instead of at the top-level namespace

File: doc/source/whatsnew/v1.1.0.rst [hunk 1]
```diff
@@ -375,7 +375,7 @@ Other API changes
 - ``loc`` lookups with an object-dtype :class:`Index` and an integer key will now raise ``KeyError`` instead of ``TypeError`` when key is missing (:issue:`31905`)
 - Using a :func:`pandas.api.indexers.BaseIndexer` with ``count``, ``min``, ``max``, ``median``, ``skew``,  ``cov``, ``corr`` will now return correct results for any monotonic :func:`pandas.api.indexers.BaseIndexer` descendant (:issue:`32865`)
 - Added a :func:`pandas.api.indexers.FixedForwardWindowIndexer` class to support forward-looking windows during ``rolling`` operations.
--
+- Exports :class:`pandas.core.index.InvalidIndexError`.
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:378]: can you add this pr number

File: pandas/__init__.py [hunk 2]
```diff
@@ -80,6 +80,7 @@
     Float64Index,
     MultiIndex,
     IntervalIndex,
+    InvalidIndexError,
```
Comment (jreback) [inline pandas/__init__.py:83]: not here, rather goes in pandas.errors

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/__init__.py (hunk 2 above)
Comment (horta) [reply pandas/__init__.py:83]: Shall then I move the definition of `InvalidIndexError` (https://github.com/pandas-dev/pandas/blob/c71bfc36211b5e2d860a06d8fbef902b757bd6e4/pandas/core/indexes/base.py#L156) to `pandas.errors` module (https://github.com/pandas-dev/pandas/blob/master/pandas/errors/__init__.py)?

Comment (jorisvandenbossche) [reply pandas/__init__.py:83]: yes, that's good

**Comment by horta** [comment]
I have now defined `InvalidIndexError` at `pandas.errors` module. Documented it (suggestions?). Most of the diff concerns importing `InvalidIndexError` from `pandas.errors`.

**Comment by mroeschke** [comment]
Could you include this error in the API reference? https://pandas.pydata.org/docs/reference/general_utility_functions.html#exceptions-and-warnings

File: pandas/errors/__init__.py [hunk 3]
```diff
@@ -200,3 +200,9 @@ class NumbaUtilError(Exception):
     """
     Error raised for unsupported Numba engine routines.
     """
+
+
+class InvalidIndexError(Exception):
+    """
+    Exception raised when attemping to use an invalid index key.
```
Comment (jreback) [inline pandas/errors/__init__.py:207]: can you add a versionadded 1.1 tag here

**Review by jreback — CHANGES_REQUESTED** [review]
small comment, ping on green.

File: pandas/core/index.py [hunk 4]
```diff
@@ -29,3 +30,5 @@
     FutureWarning,
     stacklevel=2,
 )
+
```
Comment (jreback) [inline pandas/core/index.py:33]: why is this needed?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/index.py (hunk 4 above)
Comment (horta) [reply pandas/core/index.py:33]: Do you mean exporting InvalidIndexError in here?

Comment (jreback) [reply pandas/core/index.py:33]: yes

Comment (horta) [reply pandas/core/index.py:33]: I will remove it from here.

**Comment by jreback** [comment]
@horta looks fine, can you rebase and update to the comments above, ping on green.

File: pandas/core/index.py (hunk 4 above)
Comment (horta) [reply pandas/core/index.py]: Removed.

File: pandas/errors/__init__.py (hunk 3 above)
Comment (horta) [reply pandas/errors/__init__.py:207]: Added.

**Comment by horta** [comment]
> Could you include this error in the API reference? https://pandas.pydata.org/docs/reference/general_utility_functions.html#exceptions-and-warnings

Done.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm ping on green.
